pkg/validation-webhook/snapshot.go

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"fmt"
+
+	volumesnapshotv1beta1 "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+	v1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+var (
+	// SnapshotGVR is GroupVersionResource for volumesnapshots
+	SnapshotGVR = metav1.GroupVersionResource{Group: snapshotGroup, Version: "v1beta1", Resource: "volumesnapshots"}
+	// SnapshotContentGVR is GroupVersionResource for volumesnapshotcontents
+	SnapshotContentGVR = metav1.GroupVersionResource{Group: snapshotGroup, Version: "v1beta1", Resource: "volumesnapshotcontents"}
+)
+
+// snapshot CRD api group
+const snapshotGroup = "snapshot.storage.k8s.io"
+
+// Add a label {"added-label": "yes"} to the object
+func admitSnapshot(ar v1.AdmissionReview) *v1.AdmissionResponse {
+	klog.V(2).Info("admitting volumesnapshots or volumesnapshotcontents")
+	fmt.Println("admitting volumesnapshots or volumesnapshotcontents")
+
+	reviewResponse := v1.AdmissionResponse{}
+	reviewResponse.Allowed = true
+
+	// Ignore Delete and Connect requests.
+	if !(ar.Request.Operation == v1.Update || ar.Request.Operation == v1.Create) {
```

Comment:
you wanna make the comments more accurate?
should be: "Admit requests other than Update and Create"